app/src/main/java/org/oppia/app/recyclerview/DragItemTouchHelperCallback.kt

```diff
@@ -0,0 +1,64 @@
+package org.oppia.app.recyclerview
+
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.RecyclerView
+
+/**
+ * This is used to enable functionality like  drag up and down to the RecyclerView.
+ */
+class DragItemTouchHelperCallback private constructor(dragDirs: Int, swipeDirs: Int) :
+  ItemTouchHelper.SimpleCallback(dragDirs, swipeDirs) {
+  private var dragEnabled = false
+  private lateinit var onItemDragListener: OnItemDragListener
+
+  private constructor(builder: Builder) : this(builder.dragDirs, builder.swipeDirs) {
+    dragEnabled = builder.dragEnabled
+    onItemDragListener = builder.onItemDragListener
+  }
+
+  override fun isLongPressDragEnabled(): Boolean {
+    return dragEnabled
+  }
+
+  override fun onMove(
+    recyclerView: RecyclerView,
+    source: RecyclerView.ViewHolder,
+    target: RecyclerView.ViewHolder
+  ): Boolean {
+    onItemDragListener.onItemDragged(
+      source.adapterPosition,
+      target.adapterPosition,
+      recyclerView.adapter!!
+    )
+    return true
+  }
+
+  override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {}
+  override fun onSelectedChanged(viewHolder: RecyclerView.ViewHolder?, actionState: Int) {
```

Comment:
There should be spacing between class elements per https://developer.android.com/kotlin/style-guide#vertical.

Reply:
done